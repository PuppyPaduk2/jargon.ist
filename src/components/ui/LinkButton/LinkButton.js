// @flow

import React, { type Node } from 'react';
import Link from 'gatsby-link';
import cx from 'classnames';

import '../Button/Button.scss';

type Props = {
  children: Node,
  primary?: boolean,
  className?: string,
  to: string,
};

const Button = ({ primary, className, children, ...otherProps }: Props) => (
  <Link className={cx('c-button', { 'c-button--primary': primary }, className)} {...otherProps}>
    {children}
  </Link>
);

Button.defaultProps = {
  primary: false,
  className: '',
};

export default Button;
