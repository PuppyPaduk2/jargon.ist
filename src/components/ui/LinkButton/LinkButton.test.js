// @flow

import React from 'react';
import { shallow } from 'enzyme';

import LinkButton from './LinkButton';

it('renders correctly default button', () => {
  const wrapper = shallow(<LinkButton to="/">Example button</LinkButton>);

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly primary button', () => {
  const wrapper = shallow(
    <LinkButton to="/" primary>
      Example button
    </LinkButton>,
  );

  expect(wrapper).toMatchSnapshot();
});
