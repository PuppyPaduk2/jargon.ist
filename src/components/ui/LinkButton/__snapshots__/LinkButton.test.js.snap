// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly default button 1`] = `
<GatsbyLink
  className="c-button"
  to="/"
>
  Example button
</GatsbyLink>
`;

exports[`renders correctly primary button 1`] = `
<GatsbyLink
  className="c-button c-button--primary"
  to="/"
>
  Example button
</GatsbyLink>
`;
